onnxruntime/core/framework/session_state_initializer.cc

```diff
@@ -225,27 +241,71 @@ common::Status DeserializeTensorProto(const ONNX_NAMESPACE::TensorProto& tensor_
   mlvalue.Init(p_tensor.release(),
                DataTypeImpl::GetType<Tensor>(),
                DataTypeImpl::GetType<Tensor>()->GetDeleteFunc());
-
   return common::Status::OK();
 }
 
-static common::Status PlanTensor(MLValuePatternPlanner& planner, const MLValueNameIdxMap& mlvalue_name_idx_map, const std::string& name, const ONNX_NAMESPACE::TensorProto& tensor_proto) {
-  int mlvalue_index;
-  ORT_RETURN_IF_ERROR(mlvalue_name_idx_map.GetIdx(name, mlvalue_index));
-  size_t len;
-  Status st = utils::GetSizeInBytesFromTensorProto<256>(tensor_proto, &len);
-  if (st.Code() == common::NOT_IMPLEMENTED) return Status::OK();
-  if (!st.IsOK()) return st;
-  return planner.TraceAllocation(mlvalue_index, len);
+static common::Status AllocatePlannedBuffers(const MemoryPatternGroup& mem_patterns,
+                                             const ExecutionProviders& exec_providers,
+                                             std::map<OrtAllocatorInfo, BufferUniquePtr>& weights_buffers) {
+  for (size_t i = 0; i < mem_patterns.locations.size(); i++) {
```

Comment:
>i++ [](start = 56, length = 3)

nit: for consistency, always prefer prefix increment/decrement in C++ as no copy of the object is required. makes no difference here, but if a class is involved it does.